python_scripts/cross_validation_sol_02.py

```diff
@@ -135,6 +135,38 @@
 # We see that using `strategy="stratified"`, the results are much worse than
 # with the `most_frequent` strategy. Since the classes are imbalanced,
 # predicting the most frequent involves that we will be right for the
-# proportion of this class (~75% of the samples). However, by using the
-# `stratified` strategy, wrong predictions will be made even for the most
-# frequent class, hence we obtain a lower accuracy.
+# proportion of this class (~75% of the samples). However, the `stratified`
+# strategy will randomly generate predictions by respecting the training
+# set’s class distribution, resulting in some wrong predictions even for
+# the most frequent class, hence we obtain a lower accuracy.
+#
+# Please refer to the scikit-learn documentation on [DummyClassifier
+# ](https://scikit-learn.org/stable/modules/generated/sklearn.dummy.DummyClassifier.html)
+# for more details on the meaning of the `strategy="stratified"` parameter.
+
+# %% tags=["solution"]
+dummy_most_frequent = DummyClassifier(strategy="most_frequent")
+dummy_stratified = DummyClassifier(strategy="stratified")
+n_runs = 3
+
+for run_idx in range(n_runs):
+
+    result_most_frequent = cross_validate(
```

Comment:
Since we already used `cross_val_score` in the previous notebook and we are only interested in the test score, we can simplify this example by using it.

Reply:
```python
dummy_models = {
    "Dummy 'most_frequent'": DummyClassifier(strategy="most_frequent"),
    "Dummy 'stratified'": DummyClassifier(strategy="stratified"),
}
n_runs = 3

for run_idx in range(n_runs):
    final_scores = pd.DataFrame({
        f"{name} score": cross_val_score(model, data, target, cv=cv, n_jobs=2)
        for name, model in dummy_models.items()
    })

    # ... make the plotting
```